src/service/topics.js

```diff
@@ -14,12 +15,20 @@ const getUserTopic = async (query) => {
   }
 };
 
-const putUserTopic = async (data) => {
+const putUserTopic = async (data, isIncognito) => {
+  let res;
   try {
-    const res = await api.put('/topics/follow-v2', {
-      ...data,
-      with_system_message: true
-    });
+    if (isIncognito) {
+      res = await anonymousApi.put('/topics/follow-v2', {
+        ...data,
+        with_system_message: true
+      });
+    } else {
+      res = await api.put('/topics/follow-v2', {
+        ...data,
+        with_system_message: true
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The modification in the `putUserTopic` function to conditionally use `anonymousApi` based on `isIncognito` is a good approach for handling different types of users (regular or incognito). However, error handling seems to be missing for the case when the API call fails. It would be beneficial to add some error handling logic here to provide feedback to the user or to retry the request.
```diff
  const putUserTopic = async (data, isIncognito) => {
    let res;
    try {
      if (isIncognito) {
        res = await anonymousApi.put('/topics/follow-v2', {
          ...data,
          with_system_message: true
        });
      } else {
        res = await api.put('/topics/follow-v2', {
          ...data,
          with_system_message: true
        });
      }
+     if (!res.data) {
+       throw new Error('API call failed');
+     }
      OneSignalUtil.rebuildAndSubscribeTags();
      return res.data;
    } catch (error) {
+     // Handle the error appropriately
+     console.error(error);
+     throw error;
    }
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->